src/Dialect/ONNX/ShapeInference/Pad.cpp

```diff
@@ -60,7 +60,12 @@ LogicalResult ONNXPadOpShapeHelper::computeShape(
     DimIndexExpr dimInput(dataBounds.getDim(i));
 
     // Calculation for output size.
-    IndexExpr dimOutputFinal = padBegin + dimInput + padEnd;
+    IndexExpr dimOutputFinal;
+    if (padBegin.isLiteral() && padEnd.isLiteral() &&
```

Comment:
Is the problem that if they are both zero, and the input dim is question mark, that is the only way to preserve the same question mark? 
I think we should change the propagation rules that `question mark + 0` should preserve the question mark in the index expressions. Then you can do that systematically, `? * 1`, `? / 1`, `select(true, ?, x)`...
Then we don't have this exception here, and it would apply to many many more cases.

Reply:
Yes, I am aware of this and it's interesting enough to be addressed with a new PR. So, I just do a quick fix for PadOp that is used in ResNet50.

Reply:
Sure, so add a TODO to remove this handling once the proper arithmetic is in.